apis/sharing/v1alpha1/resourceoffer_types.go

```diff
@@ -35,19 +31,24 @@ type ResourceOfferSpec struct {
 type OfferPhase string
 
 const (
+	// ResourceOfferPending indicates a pending phase, an action is required.
+	ResourceOfferPending OfferPhase = "Pending"
+	// ResourceOfferAccepted indicates an accepted offer.
 	ResourceOfferAccepted OfferPhase = "Accepted"
-	ResourceOfferRefused  OfferPhase = "Refused"
+	// ResourceOfferRefused indicates a refused offer.
+	ResourceOfferRefused OfferPhase = "Refused"
 )
 
 // ResourceOfferStatus defines the observed state of ResourceOffer.
 type ResourceOfferStatus struct {
-	// ResourceOfferStatus is the status of this ResourceOffer.
+	// Phase is the status of this ResourceOffer.
 	// When the offer is created it is checked by the operator, which sets this field to "Accepted" or "Refused" on tha base of cluster configuration.
 	// If the ResourceOffer is accepted a virtual-kubelet for the foreign cluster will be created.
-	// +kubebuilder:validation:Enum="";"Accepted";"Refused"
-	ResourceOfferStatus OfferPhase `json:"resourceOfferStatus"`
+	// +kubebuilder:validation:Enum="Pending";"Accepted";"Refused"
+	// +kubebuilder:default="Pending"
+	Phase OfferPhase `json:"phase"`
 	// VkCreated indicates if the virtual-kubelet for this ResourceOffer has been created or not.
-	VkCreated bool `json:"vkCreated"`
+	VkCreated bool `json:"vkCreated,omitempty"`
```

Comment:
Would be more verbose (i.e. VirtualKubeletCreated), for the sake of clarity. Same for the names below

Reply:
yes, and probably it will be deleted in a future pr (at least the following field) or refactored, at the moment I only added the omitempty field to make it optional and to not have to care about it in this initial pr